.devops/main-intel.Dockerfile

```diff
@@ -1,25 +1,27 @@
 ARG ONEAPI_VERSION=2024.0.1-devel-ubuntu22.04
-ARG UBUNTU_VERSION=22.04
 
 FROM intel/hpckit:$ONEAPI_VERSION as build
 
+ARG LLAMA_SYCL_F16=OFF
 RUN apt-get update && \
     apt-get install -y git
 
 WORKDIR /app
 
 COPY . .
 
-# for some reasons, "-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=Intel10_64lp -DLLAMA_NATIVE=ON" give worse performance
 RUN mkdir build && \
     cd build && \
-    cmake .. -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx && \
-    cmake --build . --config Release --target main server
+    if [ "${LLAMA_SYCL_F16}" = "ON" ]; then \
+        echo "LLAMA_SYCL_F16 is set" && \
+        export OPT_SYCL_F16="-DLLAMA_SYCL_F16=ON"; \
+    fi && \
+    cmake .. -DLLAMA_SYCL=ON -DCMAKE_C_COMPILER=icx -DCMAKE_CXX_COMPILER=icpx ${OPT_SYCL_F16} && \
+    cmake --build . --config Release --target main
 
-FROM ubuntu:$UBUNTU_VERSION as runtime
+FROM intel/hpckit:$ONEAPI_VERSION as runtime
```

Comment:
FROM intel/hpckit:$ONEAPI_VERSION as runtime
->
FROM intel/oneapi-runtime:$ONEAPI_VERSION as runtime

hpckit don't include MKL which is used by SYCL backend. Suggest to use **oneapi-runtime**.
But I don't verify it, please verify with oneapi-runtime.

Reply:
I'm pretty sure that the `intel/hpckit` is the correct one here.

The `oneapi-runtime` that you mentioned does not have compiler (icx or icpx), therefore it cannot even compile the project.

In fact, [oneapi-runtime](https://hub.docker.com/r/intel/oneapi-runtime) is stripped down version of [intel/oneapi-hpckit](https://hub.docker.com/r/intel/intel/oneapi-hpckit). I think that `oneapi-hpckit` is what you're talking about. However, `oneapi-hpckit` is being renamed to `hpckit`. You can see that info on the docker hub page.

Reply:
@NeoZhangJianyu Can you update about this? Thanks!

Reply:
I think intel/oneapi-basekit:$ONEAPI_VERSION should be sufficient for this.

Reply:
> @NeoZhangJianyu Can you update about this? Thanks!

Yes, we need icx/icpx in the project.
hpckit has no oneMKL. So I think basekit is right one.

Reply:
yeah thanks, `intel/oneapi-basekit` is the correct one. I changed the base image to `intel/oneapi-basekit`